godot-core/src/builtin/color_hsv.rs

```diff
@@ -0,0 +1,311 @@
+/*
+ * Copyright (c) godot-rust; Bromeon and contributors.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+use super::math::ApproxEq;
+use super::Color;
+
+/// HSVA floating-number Color representation.
+///
+/// Godot's [`Color`] built-in type supports mainly RGBA floating point-based notation. `ColorHsv` supports manipulating its HSVA
+/// representation by introducing conversion methods between itself and `Color`.
+///
+/// `ColorHsv` *is not* a [`GodotType`](crate::builtin::meta::GodotType). To use it in properties expecting `Color`, you need to convert
+/// it back to this type.
+///
+/// `Color` created by `Color::from(color_hsv)` is equal to one created by `Color::from_hsv()`, but the conversion time is approximately
+/// 4 times faster - partly because it avoids calls between Godot and Rust during conversion.
+///
+/// ## Conversions
```

Comment:
Maybe it would be better to have both of the methods in the `ColorHsv` namespace (as I guess we are keeping the `Color::from_hsv()` gateway to Godot builtin impl)?  `ColorHsv::from_rgba(Color) -> ColorHsv` and `ColorHsv::to_rgba(self) -> Color`

I feel like it will keep it cleaner: *if you are using methods from `ColorHsv`, you are using `gdext` custom implementation. To use Godot's builtin implementations using `gdext`, feel free to use the one on `Color`*

Reply:
No, I think it's both more symmetric and more ergonomic to have two `to_*` methods that convert each type to the other.

What is maybe a bit weird is that then `TryFrom` lives outside the inherent impl, so there's an inconsistency between `rgb.to_hsv()` and `ColorHsv::try_from(&rgb)`. A `rgb.try_to_hsv()` might be nicer and would follow our "panic + `try_*` overload" pattern we apply in many APIs. I don't really know if genericity is useful here, plus `try_into()` on its own suffers from not having type inference.